sql/errors.go

```diff
@@ -127,3 +130,15 @@ var (
 	// ErrTruncateReferencedFromForeignKey is returned when a table is referenced in a foreign key and TRUNCATE is called on it.
 	ErrTruncateReferencedFromForeignKey = errors.NewKind("cannot truncate table %s as it is referenced in foreign key %s on table %s")
 )
+
+func CastSQLError(err error) *mysql.SQLError {
+	var code int
+	var sqlState string = ""
+	switch {
+	case ErrTableNotFound.Is(err):
+		code = 1146
```

Comment:
Probably better to use the constants for these that vitess defines. They can be exported if they aren't already. Hard to review changes here with them being magic numbers.